stories/demos/exportUtils.tsx

```diff
@@ -0,0 +1,87 @@
+import { cloneElement } from 'react';
+import type { ReactElement } from 'react';
+import { renderToStaticMarkup } from 'react-dom/server';
+import XLSX from 'xlsx';
+import { jsPDF } from 'jspdf';
+import autoTable from 'jspdf-autotable';
+
+import type { DataGridProps } from '../../src';
+
+export function exportToCsv<R, SR>(gridElement: ReactElement<DataGridProps<R, SR>>, fileName: string) {
+  const { head, body, foot } = getGridContent(gridElement);
+  const content = [...head, ...body, ...foot]
+    .map(cells => cells.map(serialiseCellValue).join(','))
+    .join('\n');
+
+  downloadFile(fileName, new Blob([content], { type: 'text/csv;charset=utf-8;' }));
+}
+
+export function exportToXlsx<R, SR>(gridElement: ReactElement<DataGridProps<R, SR>>, fileName: string) {
+  const { head, body, foot } = getGridContent(gridElement);
+  const wb = XLSX.utils.book_new();
+  const ws = XLSX.utils.aoa_to_sheet([...head, ...body, ...foot]);
+  XLSX.utils.book_append_sheet(wb, ws, 'Sheet 1');
+  XLSX.writeFile(wb, fileName);
+}
+
+
+export function exportToPdf<R, SR>(gridElement: ReactElement<DataGridProps<R, SR>>, fileName: string) {
+  const doc = new jsPDF({
+    orientation: 'l',
+    unit: 'px'
+  });
+
+  const { head, body, foot } = getGridContent(gridElement);
+  autoTable(doc, {
+    head,
+    body,
+    foot,
+    horizontalPageBreak: true,
+    styles: { cellPadding: 1.5, fontSize: 8, cellWidth: 'wrap' },
+    tableWidth: 'wrap'
+  });
+  doc.save(fileName);
+}
+
+function getGridContent<R, SR>(gridElement: ReactElement<DataGridProps<R, SR>>) {
+  const grid = document.createElement('content');
+  grid.innerHTML = renderToStaticMarkup(cloneElement(gridElement, {
+    enableVirtualization: false
+  }));
+
+  return {
+    head: getRows('.rdg-header-row'),
+    body: getRows('.rdg-row:not(.rdg-summary-row)'),
+    foot: getRows('.rdg-summary-row')
+  };
+
+  function getRows(selector: string) {
+    return Array.from(
+      grid.querySelectorAll<HTMLDivElement>(selector)
+    ).map(gridRow => {
+      return Array.from(
+        gridRow.querySelectorAll<HTMLDivElement>('.rdg-cell')
+      ).map(gridCell => gridCell.innerText);
+    });
+  }
+}
+
+function serialiseCellValue(value: unknown) {
+  if (typeof value === 'string') {
+    const formattedValue = value.replace(/"/g, '""');
+    return formattedValue.includes(',') ? `"${formattedValue}"` : formattedValue;
+  }
+  return value;
+}
+
+function downloadFile(fileName: string, data: Blob) {
+  const downloadLink = document.createElement('a');
+  downloadLink.download = fileName;
+  const url = URL.createObjectURL(data);
+  downloadLink.href = url;
+  downloadLink.style.display = 'none';
+  document.body.appendChild(downloadLink);
+  downloadLink.click();
+  document.body.removeChild(downloadLink);
```

Comment:
```suggestion
  downloadLink.click();
```
I believe we can click without adding the link to the tree. Not sure if this'll work in all browsers though.

Reply:
Works on Edge, FF, and Chrome